### ISSUE TITLE
Label text getting cut off when using accessibility font scaling

### ISSUE THREAD

**Issue by ChristopherMWood** [xamarin/Xamarin.Forms#13448]
### Description
When rendering a label in XAML with font scaling turned on, in some circumstances the label text gets cut off and does not render correctly. This makes accessibility features break certain UI layouts for users depending on the text in the label. Since font scaling is not available on simulators, this must be reproduced using the test project on a physical device. Exact reproduction are for iOS iPhone 6. Issue has have been found for Android as well so this is a cross platform issue that is just easiest to reproduce in iOS due to the more options available for text size scaling.

### Steps to Reproduce
1. Pull project from https://github.com/ChristopherMWood/XamarinFontScalingIssue
3. Turn font size in iOS settings to largest size
2. Run on physical iPhone 6 (model in readme)
4. Run app and see label text cut off

### Expected Behavior
Xamarin forms label should show all text set in the text property without cutting it short

### Actual Behavior
Xamarin forms is cutting off the end text of the label when font scaled

### Basic Information

- Version with issue: Xamarin Forms 5.0.0.1874 (older versions have issue too)
- Last known good version: Unknown
- Platform Target Frameworks: 
  - iOS:  XCode 12.2 ; iOS 14.3 SDK. (has been seen on previous SDK's)
  - Android: N/A
  - UWP:  N/A
- Android Support Library / AndroidX Version: N/A
- NuGet Packages: None past default new project packages
- Affected Devices: iPhone 6 (have seen on various other devices

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
=== Visual Studio Community 2019 for Mac ===

Version 8.8.4 (build 30)
Installation UUID: ba5e56b8-68ff-4b36-b424-b48d785c7cc2
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000113

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.113 (2020-02/4fdfb5b1fd5) (64-bit)
	Package version: 612000113

=== Roslyn (Language Service) ===

3.8.0-5.20519.18+4c195c3ac1974edcefa76774d7a59a2350ec55fa

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.101/Sdks
SDK Versions:
	5.0.101
	5.0.100
	3.1.404
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.0.103
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.1
	5.0.0
	3.1.10
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.0.3
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19

=== .NET Core 3.1 SDK ===

SDK: 3.1.404

=== Xamarin.Profiler ===

Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 12.3 (17715)
Build 12C33

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 14.6.0.15 (Visual Studio Community)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:41-0500

=== Xamarin Designer ===

Version: 16.8.0.510
Hash: 44e3f3ce9
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:06:14 UTC

=== Xamarin.Android ===

Version: 11.1.0.17 (Visual Studio Community)
Commit: xamarin-android/d16-8/c0e2b8e
Android SDK: /Users/christopherwood/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 30.0.2

Build Information: 
Mono: be2226b
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/christopherwood/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:32:24 UTC

=== Android Device Manager ===

Version: 16.8.0.46
Hash: 0a81419
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:32:44 UTC

=== Build Information ===

Release ID: 808040030
Git revision: eaa21a8983aa5c4bcf6a665892478b4f3d413418
Build date: 2020-12-10 11:06:12-05
Build branch: release-8.8
Xamarin extensions: eaa21a8983aa5c4bcf6a665892478b4f3d413418

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.2.0 Darwin Kernel Version 20.2.0
    Wed Dec  2 20:39:59 PST 2020
    root:xnu-7195.60.75~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

Stack Overflow Search 0.7.1

```

</details>

### Screenshots
![iphone6Screenshot](https://user-images.githubusercontent.com/1214565/104960445-398c4480-59a2-11eb-8807-1fe6c0ba6161.png)

### Reproduction Link
Here is a repo to reproduce the issue. Details are in the readme.
https://github.com/ChristopherMWood/XamarinFontScalingIssue

### Workaround
There is no workaround that I am aware of past not using font scaling or the features affected.

**Comment by rachelkang** [comment]
Hi, @ChristopherMWood - thank for submitting this issue! I'm able to reproduce this issue on iOS, although not on Android. Would you mind sharing repro steps / device specs for how you reproduced the issue on Android?

On iOS, I'm noticing that its the specification of both a padding and FontSize (either named or numeric) results in the words getting cut off. I'm not sure what your use case is, but if removing the padding or changing the FontSize is feasible, that may be a potential workaround for now!

**Comment by ChristopherMWood** [comment]
Hey @rachelkang, thanks for reading and reproducing this issue! The test repo does not reproduce the issue on Android unfortunately. Android has way less font scaling sizes available so it's much harder to reproduce and in my testing was unable to get a consistent example that reproduced in both iOS and Android using the same layout and settings. This issue was originally reproduced in a production app I work on with way more complex layouts and components and I had to simplify it as much as possible to be able to narrow down the issue to submit this. If it is helpful, I can try to repro it in Android on my One Plus 7t phone but it will take me a little bit in between some work stuff.

As for font size, due to accessibility standards that our app must meet from being in the health care industry, we must support named sizes and users font scaling accessibility preferences. So I do not have specific control over the font size unfortunately. You are right though that when padding is removed it seems to negate the issue in many situations so we have been doing that where we can. Unfortunately for new developers that join the team, telling them not using padding "may prevent an issue" is not maintainable in the long run as a one off. We've also seen this issue pop up in tons of different random cases. This originally showed up when we created a custom button that would get rendered from text from a server. So the number of use cases is dynamic and never static. I've seen people mention issues like this on Stack Overflow and the Xamarin forums but repro is always iffy and super specific. While it's not a deal breaker, it has created a moving target when we do accessibility testing during releases so figured putting it in that context would help bring the issue some more traction.

**Comment by rachelkang** [comment]
Hi, @ChristopherMWood - thank you so much for your thorough response. I'm happy to hear that removing the padding has proven to be an effective workaround, although I agree it is not a sustainable solution in the long-term. As we work to improve accessibility for .NET MAUI, we will certainly be exploring this issue further. Please feel free to share your thoughts and ideas at https://github.com/dotnet/maui!

In the meantime, I'd love to hear any more information you're willing to share regarding your experiences with this issue. Have you ever had to develop any other odd workarounds? Are there any other related issues you've experienced with font scaling? 

Also, if you ever get to reproduce the issue on Android, I'd love to take a look at that as well!

**Comment by ChristopherMWood** [comment]
@rachelkang We have run into a random assortment of issues while developing for accessibility with font scaling but unfortunately most of them were not tracked as well as this one for the font scaling so I don't have detailed examples. If we run into them in the future I'll be sure to share them. And if I get the time I'll try to reproduce in Android, but I'm not sure when that will be with my current work load.

Thanks for sharing the link to Maui! I didn't know that was where .NET UI development was going until you shared that so it's exciting to see where the future of all of this will probably be. Also thanks in general for the follow ups and responses, it's great to see an active community around Xamarin. I'll be sure to share anything new I run into going forward.

**Comment by rachelkang** [comment]
@ChristopherMWood that's all very fair, and thanks so much! Excited to have introduced you to .NET MAUI and looking forward to seeing you on the other side! :)
